modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/InteractListener.java

```diff
@@ -70,7 +76,28 @@ public void onPlayerInteractStick(PlayerInteractEvent event) {
             if (craft == null) {
                 return;
             }
-            Long time = timeMap.get(event.getPlayer());
+            final CraftType type = craft.getType();
+            int currentGear = craft.getCurrentGear();
+            if (player.isSneaking() && !craft.getPilotLocked()) {
+                final int gearShifts = type.getGearShifts();
+                if (gearShifts == 1) {
+                    player.sendMessage(I18nSupport.getInternationalisedString("Gearshift - Disabled for craft type"));
+                    return;
+                }
+                currentGear++;
+                if (currentGear > gearShifts)
+                    currentGear = 1;
+                player.sendMessage(I18nSupport.getInternationalisedString("Gearshift - Gear changed") + " " + currentGear + " / " + gearShifts);
+                craft.setCurrentGear(currentGear);
+                return;
+            }
+            Long time = timeMap.get(player);
+            int tickCooldown = craft.getType().getTickCooldown(craft.getW());
+            if (type.getGearShiftsAffectDirectMovement() && type.getGearShiftsAffectTickCooldown()) {
+                tickCooldown *= currentGear;
+            } else if (!type.getGearShiftsAffectDirectMovement() && type.getGearShiftsAffectTickCooldown()) {
+                tickCooldown /= currentGear;
```

Comment:
What is the purpose of this?  If getGearShiftsAffectDirectMovement is false it should have no affect on the tickCooldown for direct movement.